inspirehep/modules/workflows/tasks/actions.py

```diff
@@ -615,3 +615,13 @@ def jlab_ticket_needed(obj, eng):
     jlab_categories = set(current_app.config['JLAB_ARXIV_CATEGORIES'])
     arxiv_categories = set(get_arxiv_categories(obj.data))
     return bool(jlab_categories & arxiv_categories)
+
+
+@with_debug_logging
+def load_from_source_data(obj, enj):
+    """Restore the workflow data and extra_data from source_data."""
+    source_data = obj.extra_data['source_data']
+    obj.data = source_data['data']
+    obj.extra_data = source_data['extra_data']
+    obj.extra_data['source_data'] = deepcopy(source_data)
+    obj.save()
```

Comment:
Throw exception with nice message pointing to dev work if there's no source_data (so the curators don't go crazy restarting the workflow)

Reply:
Copy over the _task_history until we have proper logs :cry: